### ISSUE TITLE
System files replacement

### ISSUE THREAD

**Issue by elkarto** [350d/70Mai_A800S_Firmware#7]
Inspecting the code of the script to replace the logo, I see that what it does is to replace a file in the camera's file system.

Can this replacement be done via ftp at any time? Or does it have to be done during camera startup and that's why a script is used?

I was thinking of replacing the audio files, to make ADAS less annoying, and to fix some mistranslation into my language.

It might also be interesting to replace some of the icons.

**Comment by 350d** [comment]
Logo image located in `/misc` folder and its read/write folder and can be updated via ftp any time (just keep proper permissions on file). Audio files located on virtual partition an dits read only. You can only patch it before pack firmware back.

**Comment by 350d** [comment]
ADAS patched with beep signals in latest version for English audio

**Comment by elkarto** [comment]
Just thinking,

It would be possible to replace the audio files (or other types) in the virtual partition with symlinks before packing the firmware, pointing to a memory area that is read/writeable??

To later replace the files with whatever we want.


I think this way it would be easier and more practical.


I have no experience with linux filesystems and maybe this is a nonsense and cannot be done.

**Comment by elkarto** [comment]
Anyway, I've been looking at how to package my own firmware thanks to your help in  #1 and I think I will give it a try.


Anything to keep in mind so I don't end up with a useless paperweight?

**Comment by 350d** [comment]
Keep OTA FW nearby to restore your device in case of custom FW failure. Its almost impossible to brick this device anyway.

**Comment by elkarto** [comment]
Is it possible to use your firmware as origin? Or is it necessary to use the factory firmware for that?

It would be interesting to share our projects (i.e. the patch folder) (if you are open to it, of course, it's your work and your time)

**Comment by 350d** [comment]
You can use custom too. Its works fine for me.

**Comment by elkarto** [comment]
I believe I have managed to set up the dev environment without problems.

I don't have the camera here so I can't test if the generated firmware is valid.


The next thing worries me:

- I unpack the firmware without errors.
- I repack the firmware, no modifications, no patches, and no errors.

The output file and the source file are not binary-same.



Is this a normal behaviour?

I will check if the resulting firmwares are valid when I have the camera with me.

**Comment by 350d** [comment]
If you see no errors in create log - you good.

**Comment by elkarto** [comment]
It works!

I have managed to change the ADAS audios in Spanish language.

I have also changed some UI icons and not much more.


Other things at the moment are too complex for me, I will try to compare your unpacked firmware with the original one, and see if I can learn something about how things work under the hood.


Next step, investigate how the camera's cgi-bin API works. Have you done any tests on it?


Thanks for your help!!
